models/base.py

```diff
@@ -890,45 +854,35 @@ def save(self, data, extension=None, overwrite=True, exists_ok=True, verify_md5=
 
         if mustupload:
             remmd5 = self.archive.upload( localpath, relpath.parent, relpath.name, overwrite=overwrite, md5=origmd5 )
-            remmd5 = UUID( remmd5 )
             if curextensions is not None:
-                extmd5s[extensiondex] = remmd5
+                extmd5s[extension] = remmd5
                 self.md5sum = None
                 self.filepath_extensions = curextensions
                 self.md5sum_extensions = extmd5s
             else:
-                self.md5sum = remmd5
+                self.md5sum = UUID( remmd5 )
 
-    def remove_data_from_disk(self, remove_folders=True, purge_archive=False, session=None, nocommit=False):
+    def remove_data_from_disk(self, remove_folders=True, purge_archive=False):
 
         """Delete the data from disk, if it exists.
         If remove_folders=True, will also remove any folders
         if they are empty after the deletion.
 
+        It is the responsibility of the caller to make sure
+        these changes (i.e., the nullifying of the md5sums)
+        is also saved to the DB, or this entire object can be deleted
+        If not, there would be database objects without a corresponding
+        file on local disk / archive and that's not good!
+
```

Comment:
I'm worried that there's a session in this function. Whenever we need to pass a session in it makes it more likely that we will be opening sessions we didn't mean to. Not to mention that now you have to worry about not committing. 

This way you have a clean removal of files without having side effects that you may not want to have. The problem of having missing files is non-trivial, but that's true anyway because you can always just take the filename and delete the file manually.

Reply:
I will add another function `delete_from_disk_and_database` that accepts `session` and `commit` arguments. 

I think if we want to not surprise the user (and ourselves) we should keep the `remove_file_from_disk` to do exactly just that and not other things. That way you get to choose from three explicit options:
(1) delete just the files
(2) delete and remove from database using your own session
(3) delete and remove from database using internal session (by not passing a session)
and you can control the commit step or not, which I think should be off by default.

Reply:
I changed my mind, `commit` should be true by default because if you don't give a session then the internal session will close without committing by default, and that's not what we want. 
I would also raise an exception if `session=None` and `commit=False` were used at the same time.

Reply:
Given delete_from_disk_and_database, I don't think the purge_archive functionality is needed here anymore.  This becomes the function for cleaning up the local data storage, and the other one for removing things altogether.

I'm not sure there's ever a case where we'd want to remove something from the archive but not the database.